### DISCUSSION TITLE
Tabler Icons

### DISCUSSION THREAD

**Discussion by MGeurts** [tasteui/tasteui#392] [Ideas]
Just in case there's need for another icon stack: [https://tabler.io/icons](https://tabler.io/icons).
They add new icons frequently ald have a large fanbase.

**Comment by devajmeireles** [comment]
Could you do a test for us, please? Verify that the SVG transformed into an anonymous Blade component can receive TailwindCSS properties such as size and colors. If yes, we can add this as a new option to the icon library in TallStackUI.

For example, download and transform the `abacus` into a blade component, set the TailwindCSS properties, and use the component. Finally, check if the TailwindCSS properties will be correctly applied.

`class="w-5 h-5 text-red-500"`

**Comment by MGeurts** [comment]
I found this discussion : [https://github.com/tallstackui/tallstackui/issues/140](https://github.com/tallstackui/tallstackui/issues/140)

I will try out phosphoricons first. Maybe they're as good or even better as Tabler Icons.

**Reply by devajmeireles** [reply]
The reference you cited was a topic discussed in the past when we only supported a single icon library, Heroicons. We currently support three and can add others if we identify that it works well. I'm going testing the Tabler.

**Reply by MGeurts** [reply]
@devajmeireles 
Just a quick question regarding Phosphor Icons...
I noticed that only the older icon are available in TallStackUI.
How can we update the dependency to include the latest icons, like [these ones](https://phosphoricons.com/?q=%22network-x%22&weight=%22regular%22) ?

Thanks.

**Comment by MGeurts** [comment]
So for the last 10 days, I tried out Phosphor Icons in a bigger project.
Allthough Phosphor Icons has 9,072  icons, I'm missing a lot of icons available in Tabler Icons (5,200 icons).
Phosphor Icons often has more versions of the same icon, but Tabler Icons is more divers and thematically better organized.
So I'm going to migrate back to Tabler Icons.
I think Tabler icons is a good candidate for being the fourth icon library to support.
Thanks.

**Reply by MGeurts** [reply]
You asked me to test if Tabler Icons SVG transformed into an anonymous Blade component can receive TailwindCSS properties such as size and colors. I can confirm they do. I use slightly different way to use the Tabler icons in my projects.

1. A dedicated blade component **\resources\views\components\icon\tabler.blade.php** :

`<svg viewBox="0 0 24 24" stroke="currentColor" stroke-width="{{ $strokeWidth ?? 2 }}" class="inline-block relative size-5 {{ $class ?? '' }}" {{ $attributes->except(['class', 'icon']) }}>
    <use xlink:href="/tabler/tabler-sprite-nostroke.svg#tabler-{{ $icon }}" />
</svg>`

2. Using the dedicated blade component in normal blade files :

`<x-icon.tabler icon="brand-linkedin" class="size-6 text-neutral-900 dark:text-neutral-200" />`
`<x-icon.tabler icon="{{ $sibling->sex == 'm' ? 'gender-male' : 'gender-female' }}" />`
`<x-icon.tabler icon="binary-tree" class="me-2 rotate-180" />`
`<x-icon.tabler icon="hearts" class="text-emerald-600" />`

The dedicated blade component references the icon by its **name** property from a **tabler-sprite-nostroke.svg** available in the [Tabler Icons NPM package](https://www.npmjs.com/package/@tabler/icons) and stored localy in the public folder of the webserver. Only caveat: I have to renew the **tabler-sprite-nostroke.svg** file every time I want to use icons from a **newer release**.

**Reply by devajmeireles** [reply]
Thanks! I'll include this in the TallStackUI.

**Comment by devajmeireles** [comment]
Hey, @MGeurts ! We added the Tabler icons, but I rolled it back when I realized that Tabler doesn't have the same amount of icons between outline and fill. Could you help us by suggesting other libraries that contain a good amount of icons? Preferably those that contain the same number of icons between types.

**Reply by MGeurts** [reply]
An icon can only be filled when its outlines are 100% enclosed.
That's why there is such a big gap between the number of outlined and filled icons.

Would it be a solution to take into account **only the outlined icons**?

**Comment by devajmeireles** [comment]
Thanks, @MGeurts ! We just released version 1.24.0 with support for Tabler icons outline style.

**Comment by MGeurts** [comment]
Thank you AJ, for your hard work crafting this beautiful package.
